File: radix_sort.py
def radixsort(A):
    l=len(A)
    maxelement=max(A)
    digits=len(str(maxelement))
    a=[]
    bins=[a]*10
    for i in range(digits):
        for j in range(l):
            e=int(A[j]/pow(10,i))%10
            if len(bins[e])>0:
                bins[e].append(A[j])
            else:
                bins[e]=[A[j]]
        k=0
        print("bins",bins)
        for x in range(10):
            if len(bins[x])>0:
                for y in range(len(bins[x])):
                    A[k]=bins[x].pop(0)
                    k=k+1


A=[23,45,564,28,976,1,234]
print("original",A)
radixsort(A)
print("sorted",A)

 

File: radix_sort.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func radixSort(A []int) {
	n := len(A)
	maxnum := A[0]
	for _, v := range A {
		if v > maxnum {
			maxnum = v
		}
	}

	digits := len(strconv.Itoa(maxnum))
	bins := make([][]int, 10)
	fmt.Println(bins)

	for i := 0; i < digits; i++ {
		for j := 0; j < n; j++ {
			e := (A[j] / int(math.Pow(10, float64(i))) % 10)
			bins[e] = append(bins[e], A[j])
		}

		k := 0

		for x := 0; x < 10; x++ {
			if len(bins[x]) > 0 {
				for len(bins[x]) > 0 {
					A[k] = bins[x][0]
					bins[x] = bins[x][1:]
					k++
				}
			}
		}
	}

}

func main() {
	A := []int{63, 250, 835, 947, 651, 28}

	fmt.Println("Original Array:", A)
	radixSort(A)
	fmt.Println("Sorted Array:", A)
}
